src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -37,28 +37,32 @@
   --source_file_ext SOURCE_FILE_EXT
                         Select files from source data directory that match the given extension.
                         
-Note: Based on our experiment with GTDB-Tk, we have determined that the optimal chunk size is 1000 genomes.
-With 4 batches running in parallel, each using 32 cores, it takes around 50 minutes to process a single chunk.
-To reflect this, we have set the "threads" and "program_threads" parameters in "_create_task_list" to 32, 
-indicating that each batch will use 32 cores. We have also set the execution time for GTDB-Tk to 65 minutes, 
-with some additional buffer time. To allow for a sufficient number of batches to be run within a given time limit, 
-we have set the "NODE_TIME_LIMIT_DEFAULT" to 5 hours. With these settings, we expect to be able to process up to 16 batches, 
-or 16,000 genomes, per node within the 5-hour time limit. We plan to make these parameters configurable based on 
-the specific tool being used. After conducting performance tests, we found that utilizing 32 cores per batch and 
-running 4 batches in parallel per NERSC node resulted in optimal performance, despite each node having a total of 
-256 cores.
 '''
 
 TOOLS_AVAILABLE = ['gtdb_tk', 'checkm2', 'microtrait', 'mash', 'eggnog']
 
-# estimated execution time (in minutes) for each tool to process a chunk of data
+NODE_TIME_LIMIT_DEFAULT = 5  # hours  # TODO: automatically calculate this based on tool execution time and NODE_THREADS
```

Comment:
How can you calculate this and num nodes? If you have to calculate both you're under constrained I think

Reply:
yea. You are correct. The `todo` was blindly copied. It's no longer valid.

Reply:
:+1: